### DISCUSSION TITLE
How to make Select filter in ProductResource base on  catagory type ?

### DISCUSSION THREAD

**Discussion by jigujigu** [filamentphp/filament#9591] [Help] [answered]
How to make Select filter in ProductResource base on catagory type field.

In my catagory model, I has 'HasMany relationship' to product. 
And in my product model, I has 'belongTo relationship' to catagory.

The type field in catagory is string. and the value can be  1,2,3
F=Finish Good, M=Material, H=Half-Finish

```
Table Catagory            | Table Product
id  name      type        | id   name   catagory_id
1   Class-A   F           | 1    AAA    1
2   Class-B   M           | 2    BBB     1
3   Class-C   M           | 3    CCC    2 
4   Class-D   F           | 4    DDD    4
5   Class-E   H           |5    ABC     3
...etc                    | ....etc
```
I need to make Select filter in ProductResource and has options all available type in catagory : 
'Finish Good','Material','Half-Finish'

Below is my select and not work as expected
```
SelectFilter::make('type')->label('type')
   ->options( Catagory::distinct()
   ->pluck('type', 'type'))
```

Need advice.
Thanks

**Comment by leandrocfe** [comment, marked as answer]
this?

```php
SelectFilter::make('type')
    ->options([
        'F' => 'Finish Good',
        'M' => 'Material',
        'H' => 'Half-Finish',
    ])
    ->query(fn (Builder $query, array $data) => $query->when($data['value'],
        fn (Builder $query, $type): Builder => $query->whereRelation('category', 'type', $type)),
    )
```
You could use [enums](https://filamentphp.com/docs/3.x/support/enums#using-the-enum-label-with-form-field-options) as options
